pages/blog/building-async-flight-notification-service.md

```diff
@@ -0,0 +1,257 @@
+
+## Monitoring flight status information 
+
+The Monitor service checks the status of the user’s flight by calling the On-Demand Flight Status API, which provides real-time flight schedule information like departure/arrival times, gate, or terminal. A simple cURL request to the API shows how the information is represented:
+
+        curl https://test.api.amadeus.com/v2/schedule/flights?carrierCode=KL&flightNumber=1772&scheduledDepartureDate=2021-02-18  
+
+In the JSON response, the schedule data of this example has one single segment (a leg of an itinerary, in airline jargon) with several `flightPoints`:
+
+```json
+"flightPoints": [  
+    {  
+        "iataCode": "FRA",  
+        "departure": {  
+            "terminal": {  
+                "code": "1"  
+            },  
+            "gate": {  
+                "mainGate": "B20"  
+            },  
+            "timings": [  
+                {  
+                    "qualifier": "STD",  
+                    "value": "2020-11-05T18:20+01:00"  
+                }  
+            ]  
+        }  
+    },  
+    {  
+        "iataCode": "AMS",  
+        "arrival": {  
+            "terminal": {  
+                "code": "1"  
+            },  
+            "gate": {  
+                "mainGate": "A04"  
+            },  
+            "timings": [  
+                {  
+                    "qualifier": "STA",  
+                    "value": "2020-11-05T19:35+01:00"  
+                }  
+            ]  
+        }  
+    }
+]
+```
+
+We can see that:  
+
+- The flight is scheduled to depart from Terminal 1, Gate B22 of Frankfurt International Airport (FRA) at 18:20 (UTC+1). 
+- It is scheduled to arrive at Terminal 1, Gate A04 of Amsterdam Schiphol Airport (AMS) at 19:35 (UTC+1). 
+
+The API is synchronous and therefore needs to be polled to monitor the flight status. This isn’t ideal and we need a solid strategy to avoid DDoSing the Amadeus backend, using up our free call quota or piling up a massive bill at the end of the month. 
+
+To solve this, we put the Monitor service on a separate thread. Every five minutes, the thread checks to see if it’s time to retrieve information from the API and update the status. The Monitor only calls the API if two conditions are met: 
+
+- The current date is equal to the departure date. 
+- The current time is within 4 hours of the departure time. 
+
+## Subscribing to flight updates 
+
+The Subscriber service  lets users subscribe to the notifications. We built a simple HTTP server with Flask to let the user enter their name, phone number and flight information.
+
+Once it receives this information, the Subscriber emits a `flightQueue` message with that information in the payload to the broker, so that it can be received by the Monitor.
```

Comment:
`Once it receives this information` what do you mean? Subscriber service has REST POST endpoint where it gets info about new users, new notification subscribers?

maybe better

`Once the Subscriber service gets a new user subscription, it emits a `flightQueue` <and the rest>`

?